.semaphore/semaphore.yml

```diff
@@ -0,0 +1,42 @@
+version: v1.0
+name: "Semaphore pipeline for alfred-margaret"
+
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu1804
+
+# Automatically cancel jobs that are already running for all branches but master
+auto_cancel:
+  running:
+    when: "branch != 'master'"
+
+blocks:
+  - name: Stack
+    task:
+      prologue:
+        commands:
+          # Checkout out the repository
+          - checkout
+          - git status
+          # Restore .stack-work and ~/.stack from cache for faster builds
+          - cache restore home-stack-$SEMAPHORE_GIT_BRANCH
+          - cache restore stack-work-$SEMAPHORE_GIT_BRANCH
+          # Install stack, upgrade it to latest version and add it to PATH
+          - sudo apt-get install haskell-stack
+          - stack upgrade
```

Comment:
For better reproducibility, you should also pass `--binary-version` so that it doesn't randomly upgrade to a new version with breaking changes suddenly.

And you should probably also pass `--binary-only` so that it doesn't try building stack from source in certain circumstances (though not sure when that would happen, but the docs also seem to suggest to pass this).

Reply:
Makes sense, I will add those.